src/Models/SystemMetric.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Taecontrol\MoonGuard\Models;
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Casts\Attribute;
+use Illuminate\Database\Eloquent\Factories\Factory;
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+use Taecontrol\MoonGuard\Repositories\SiteRepository;
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Taecontrol\MoonGuard\Database\Factories\SystemMetricFactory;
+
+class SystemMetric extends Model
+{
+    use HasFactory;
+
+    const CREATED_AT = 'created_at';
+
+    protected $fillable = [
+        'cpu_usage',
+        'memory_usage',
+        'disk_usage',
+        'site_id',
+    ];
+
+    public function site(): BelongsTo
+    {
+        return $this->belongsTo(SiteRepository::resolveModelClass());
+    }
+
+    public function diskUsage(): Attribute
+    {
+        return Attribute::make(
+            set: function ($value) {
+                $diskUsage = json_decode($value, true);
+                $freeSpace = $diskUsage['freeSpace'] ?? null;
+                $totalSpace = $diskUsage['totalSpace'] ?? null;
+
+                $diskPercentage = $totalSpace ? ($totalSpace - $freeSpace) / $totalSpace * 100 : 0;
+
+                return number_format($diskPercentage, 2);
+            }
+        );
+    }
```

Comment:
After making the previous change you may need to change this code into:

```suggestion
    public function diskUsage(): Attribute
    {
        return Attribute::make(
            get: function ($value) {
                $freeSpace = $value['freeSpace'] ?? null;
                $totalSpace = $value['totalSpace'] ?? null;
                $percentage = 0
                
                if($totalSpace && $freeSpace) {
                    $percentage = number_format(($totalSpace - $freeSpace) / $totalSpace * 100, 2);
                }

                return [
                    'freeSpace' => $freeSpace,
                    'totalSpace' => $totalSpace,
                    'percentage' => $percentage,
                ];
            }
        );
    }
```

Reply:
1), we've to check if freeSpace is not null too.

2) you have to use `get` instead of `set`.

If you use `set` you're storing the percentage value into the database. But we're losing the data of freeSpace and totalSpace, so let's store the json that comes from the request and on the laravel codebase when we need to use disk_usage we can add a new param into disk_usage array, then you can use it in the code like:

```php
$systemMetric->disk_usage['percentage'];
//or
data_get($systemMetric->disk_usage, 'percentage')
```

Reply:
@rpereira-tae but to use `$systemMetric->disk_usage['percentage'];`  inside the function `checkLimits` I have to use `$systemMetric = SystemMetric::where('site_id', $site->id)->first();` in order to use it?

Reply:
In the controller when you create the record it returns the model instance.

```php
$systemMetric = SystemMetric::create([
            'cpu_usage' => $request->input('cpuLoad'),
            'memory_usage' => $request->input('memory'),
            'disk_usage' => json_encode($request->input('disk')),
            'site_id' => $site->id,
        ]);
        
//then you can pass the percentage value into the checkLimits function with $sytemMetric->disk_usage['percentage'];
```

Reply:
I made a mistake on my last review, you have to keep the mutator as follow:

```suggestion
    public function diskUsage(): Attribute
    {
        return Attribute::make(
            set: function ($value) {
                $freeSpace = $value['freeSpace'] ?? null;
                $totalSpace = $value['totalSpace'] ?? null;
                $percentage = 0
                
                if($totalSpace && $freeSpace) {
                    $percentage = number_format(($totalSpace - $freeSpace) / $totalSpace * 100, 2);
                }
                return [
                    'freeSpace' => $freeSpace,
                    'totalSpace' => $totalSpace,
                    'percentage' => $percentage,
                ];
            }
        );
    }
```

This way every time you store the `disk_usage` from Larvis the percentage field will be generated and stored too on the `disk_usage` columns in db. You should clean this table records on DB